src/main/java/com/blog/web/services/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,66 @@
+package com.blog.web.services.impl;
+
+import com.blog.web.dto.RegistrationDto;
+import com.blog.web.models.Role;
+import com.blog.web.models.UserEntity;
+import com.blog.web.repository.RoleRepository;
+import com.blog.web.repository.UserRepository;
+import com.blog.web.security.SecurityUtil;
+import com.blog.web.services.UserService;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+import java.util.Arrays;
+
+@Service
+public class UserServiceImpl implements UserService {
+    final private UserRepository userRepository;
+    final private RoleRepository roleRepository;
+    final private PasswordEncoder passwordEncoder;
+
+    public UserServiceImpl(
+            UserRepository userRepository,
+            RoleRepository roleRepository,
+            PasswordEncoder passwordEncoder) {
+        this.userRepository = userRepository;
+        this.roleRepository = roleRepository;
+        this.passwordEncoder = passwordEncoder;
+    }
+
+    @Override
+    public void saveUser(RegistrationDto registrationDto) {
+        UserEntity user = new UserEntity();
+        user.setUsername(registrationDto.getUsername());
```

Comment:
i see alot of repitition when you instantiate Users consider doing a builder or a constructor with params -> I'm assuming blank Users aren't "legal" so don't let it be